wicker/core/storage.py

```diff
@@ -309,9 +332,9 @@ def _get_dataset_assets_path(self, dataset_id: DatasetID, prefix_to_trim: Option
             dataset_id.version,
             "assets",
         )
-        if prefix_to_trim:
-            return full_path.replace(prefix_to_trim, "")
-        return full_path
+        return self.__format_pathing(
+            input_path=full_path, prefix_replace_path=self._prefix_replace_path, prefix_to_trim=prefix_to_trim
```

Comment:
Sorry why we are passing `_prefix_replace_path` as class variable while `prefix_to_trim` is an arg to all the function that use `self.__format_pathing`? shouldn't they passed together?

Reply:
No because we want to have one single definition of what you are replacing given that there should be only one Wicker root path. If we passed them both the users would theoretically be able to change that path and break, the logic would inconsistent with the expected performance.

Reply:
but why that reasoning apply differently to `prefix_to_trim` that is passed in each of the functions, vs `_prefix_replace_path` that is passed as class variable?

Reply:
The use case is that a user specifies the mount point location on the drive as the `self._prefix_replace_path`. In s3mountpoint you can mount a drive and it loses some posix features but those aren't needed. So here a user can specify that there drive exists at `<drive-location>` by replacing that path with the prefix replace path. The reasoning that they are treated differently is that I don't want the internal logic of these functions to change drastically or for older versions to not be compatible with the changes.

If they are passed together and you upgrade to this release version you will be required to go and change the function calls you are using. So instead we provide a class variable that effectively says, `at every occurence of whatever my trim prefix is I want to replace it with the mountpoint path`. This boils down to saying, "I mounted my bucket at `<file-path>` and now I replace `s3://` or whatever with the path to said mounted bucket on the local system.

Reply:
Maybe I misunderstood the question and Chris question pointed out the misunderstand. I want it different so that we can do a single replace in that function and not have to have a second conditional in order to replace it with "" if the value is None